### PR TITLE
Added menu page gallery animation.

### PR DESCRIPTION
**Pull request by Aurazor**
Added menu page gallery animation
gallery_animation

### CODE REVIEW

**Comment by Saamiyah** [comment]
Looks good, I feel like 7s is a bit long for the animation to change, I find 3-4s better.

File: src/components/MeetupGallery.astro [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      for(let i = 0; i < size; i++) {
+        let arr = [];
+        arr = array.splice(0, splitSize);
+        resultArray.push(arr);
+      }
+      return resultArray;
+    }
+  </script>
+
+  <style lang="scss">
```
Comment (MrSunshyne) [inline src/components/MeetupGallery.astro:116]: Please do not add sass as a new dependency. Let's keep it css with css variables where needed. We can however use post-css nested for the nesting feature.

File: src/components/MeetupGallery.astro [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        display: grid;
+        place-items: center;
+
+        .container{
+            width: 100px;
+            height: 100px;
+            position: relative;
+
+            .photo-container{
+                width: 231px;
```
Comment (MrSunshyne) [inline src/components/MeetupGallery.astro:128]: use a css variable for the size of the container
`--photoSize: 231px`
width: var(--photoSize: 231px);
This lets you do the responsive easily and with less code.

File: src/components/MeetupGallery.astro [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                height: 240px;
```
Comment (MrSunshyne) [inline src/components/MeetupGallery.astro:129]: Can the width and height be the same?

File: src/components/MeetupGallery.astro [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                //   class=""
+                  loading="lazy"
+                />
+              </div>
+            ))
+        }
+      </div>
+  </div>
+
+  <script define:vars={{ props }}>
```
Comment (MrSunshyne) [inline src/components/MeetupGallery.astro:25]: very nice use of define:vars 👏

File: src/components/MeetupGallery.astro [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+
+    const imageArray = getSplitArray(props.currentAlbum, 3);
+    let counters = [0, 0, 0];
+    let isMouseOver = false;
+
+
+    startAnimation();
```
Comment (MrSunshyne) [inline src/components/MeetupGallery.astro:32]: It seems odd that you call these functions before they are declared. 
Perhaps they should be at the end of the script

File: src/components/MeetupGallery.astro [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    addEventListenerOnImages();
+
+    function startAnimation(){
+      setInterval(() => {
+        if(!isMouseOver){
+          changeImages();
+        }
+      }, 7000);
```
Comment (MrSunshyne) [inline src/components/MeetupGallery.astro:40]: move the duration to a variable, or even better,a prop that can be passed to the component from astro. if no prop is passed set a default value. probably 3000ms

**Review by MrSunshyne — CHANGES_REQUESTED** [review]

**Comment by kushul** [comment]
Good job so far @Aurazor

**Pull request merged**